packages/harpguru-core/src/styles/sizescheme/use-sizes.ts

```diff
@@ -0,0 +1,135 @@
+import { useGlobal } from 'reactn'
+
+import { SizeScheme, SizeSchemes } from '../styles-types'
+import { getWindowDimensions } from '../../packages/get-window-dimensions'
+
+// Here's what we need this hook to provide:
+// 1. A set of sizes which updates as the number of columns being presented increases (which is also *possibly* sensitive to the number of rows present too)
+// 2. A set of sizes which do not update as the number of columns being presented increases
+//
+// 1. is for the harpface, the left hand legend and the menu tabs. I want all of these to be harmoneously sized together. If you have the eyesight or the screen size to present all the holes at the same time then you can afford the side panel info to scale along with it
+// 2. is for the menu contents. The text is already large, and nicely so. We don't need it to be bigger or smaller, and having it scale is not the same harmoneous thing as the main page. I want the guttering to move with the main page though so that the left menu items are
+// never obscuring what you can see in the left hand legend
+// 2. may also need to be used in the quiz mode.. it looks like the quiz mode might be behaving strangely now.. the question isn't always displayed. Perhaps that's caused by something else, but it doesn't make sense that the quiz question or the other notifications for that matter should be scaled.
+
+const relativeSizes: Omit<
+  SizeScheme,
+  | 'columnWidth'
+  | 'rowHeight'
+  | 'fragmentGutter'
+  | 'labelProtrusion'
+  | 'labelIconSize'
+> = {
+  0: 0,
+  1: 1,
+  2: 1.618,
+  3: 2.618,
+  4: 4.236,
+  5: 6.854,
+  6: 11.09,
+  7: 17.944,
+  8: 29.034,
+  9: 46.979,
+  10: 76.013,
+  11: 122.989,
+} as const
+
+const relativeColumnWidth = 9
+const relativeFragmentGutterWidth = 7
+const relativeLabelProtrusion = 9
+const relativeLabelIconSize = 7
+
+export const useSizes = (): SizeSchemes => {
+  const { shortEdge, longEdge } = getWindowDimensions()
+
+  const [activeHarpStrata] = useGlobal('activeHarpStrata')
+  const { degreeMatrix } = activeHarpStrata
+  const { [0]: exampleHarpRow } = degreeMatrix
+  const { length: harpRowCount } = degreeMatrix
+  const { length: harpColumnCount } = exampleHarpRow
+
+  const {
+    [relativeColumnWidth]: columnWidth,
+    [relativeFragmentGutterWidth]: fragmentGutter,
+    [relativeLabelProtrusion]: labelProtrusion,
+    [relativeLabelIconSize]: labelIconSize,
+  } = relativeSizes
+  const rowHeight = columnWidth
+  const labelGrace = fragmentGutter
+
+  // TODO: It might be better to either count the number of gutters
```

Comment:
TODO here

Reply:
Raised as techdebt here: https://github.com/js-jslog/harpguru/issues/113#issuecomment-922036125

The work done to make the app sizing dynamic (at #107) was used as a platform for additional speculative pull requests to ensure that this was the correct direction to start down.

Rather than addressing the finer details in this or subsequent pull request and potentially struggling to merge up to those subsequent branches I will prefer to record the issues in each merge request and resolve them afterwards.

This has a limit of course. Major issues will not be merged to master. I will need to find a way around that if it happens.